src/file_operations.rs

```diff
@@ -86,6 +89,14 @@ impl FileOperations {
         self.operations.push(Op::MkDirAll(self.root.join(name)))
     }
 
+    pub fn link<P: AsRef<Path>>(&mut self, source: P, dest: P) {
```

Comment:
I'm unsure if this is something we need to worry about, but I think the way this generic is defined will mean that we always have to pass the same type of arguments for both source and dest.

This might make it kind of awkward to use in certain situations.  The solution would be to have two generic params of the same type with different names.